datafusion/expr/src/function.rs

```diff
@@ -84,6 +84,72 @@ impl<'a> AccumulatorArgs<'a> {
     }
 }
 
+/// `StateFieldsArgs` encapsulates details regarding the required state fields for an aggregate function.
+///
+/// - `name`: Name of the aggregate function.
+/// - `input_type`: Input type of the aggregate function.
+/// - `return_type`: Return type of the aggregate function. Defined by `fn return_type` in AggregateUDFImpl.
+/// - `nullable`: Indicates whether the field can be null.
+pub struct FieldArgs<'a> {
+    pub name: &'a str,
+    pub input_type: &'a DataType,
+    pub return_type: &'a DataType,
+    pub nullable: bool,
+}
+
+impl<'a> FieldArgs<'a> {
```

Comment:
Given all the fields are pub, I wonder how much benefit this API adds over simply creating FieldArgs directly 🤔

Reply:
But like `name`, it includes arguments, not every time is the same, how do they get the correct field without `args.name`?